### PR TITLE
prefactor to remove unnecessary usages of XFormPillow

### PR DESCRIPTION
**Pull request by snopoke**
@emord 

best reviewed by commit :fish:

### CODE REVIEW

File: corehq/elastic.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 from corehq.pillows.mappings.reportcase_mapping import REPORT_CASE_INDEX
 from corehq.pillows.mappings.sms_mapping import SMS_INDEX_INFO
 from corehq.pillows.mappings.user_mapping import USER_INDEX_INFO
-from corehq.pillows.mappings.xform_mapping import XFORM_INDEX
+from corehq.pillows.mappings.xform_mapping import XFORM_INDEX, XFORM_INDEX_INFO
```
Comment (dimagimon) [inline corehq/elastic.py]: F401 'XFORM_INDEX' imported but unused

File: corehq/pillows/reportxform.py [hunk 2]
```diff
@@ -20,18 +21,21 @@ class ReportXFormPillow(XFormPillow):
...
+    doc_ret = transform_xform_for_elasticsearch(doc_dict)
+    if doc_ret:
+        domain = doc_dict.get('domain', None)
 
-            return doc_ret
-        else:
+        if domain not in getattr(settings, 'ES_XFORM_FULL_INDEX_DOMAINS', []):
+            # full indexing is only enabled for select domains on an opt-in basis
             return None
+        convert_property_dict(doc_ret['form'], REPORT_XFORM_MAPPING['properties']['form'], override_root_keys=['case'])
```
Comment (dimagimon) [inline corehq/pillows/reportxform.py]: E501 line too long (119 > 115 characters)

File: corehq/apps/data_analytics/tests/test_malt_generator.py [hunk 3]
```diff
@@ -11,11 +11,12 @@
...
 from corehq.apps.smsforms.app import COMMCONNECT_DEVICE_ID
 from corehq.const import MISSING_APP_ID
-from corehq.pillows.xform import XFormPillow
+from corehq.elastic import get_es_new
+from corehq.pillows.mappings.xform_mapping import XFORM_INDEX_INFO
 from corehq.util.elastic import ensure_index_deleted
 
 from dimagi.utils.dates import DateSpan
-from pillowtop.es_utils import completely_initialize_pillow_index
+from pillowtop.es_utils import completely_initialize_pillow_index, initialize_index_and_mapping
```
Comment (dimagimon) [inline corehq/apps/data_analytics/tests/test_malt_generator.py]: F401 'completely_initialize_pillow_index' imported but unused

**Comment by snopoke** [comment]
Seems it got merged with a broken build.

**Pull request merged**